workbench-core/infrastructure/src/workbenchCognito.test.ts

```diff
@@ -165,7 +190,35 @@ describe('WorkbenchCognito tests', () => {
     });
 
     // User Pool Client
-    template.resourceCountIs('AWS::Cognito::UserPoolClient', 1);
+    template.resourceCountIs('AWS::Cognito::UserPoolClient', 2);
+
+    template.hasResourceProperties('AWS::Cognito::UserPoolClient', {
+      AllowedOAuthFlows: ['code'],
+      AllowedOAuthFlowsUserPoolClient: true,
+      AllowedOAuthScopes: ['openid'],
+      CallbackURLs: workbenchCognitoProps.websiteUrls,
+      EnableTokenRevocation: true,
+      GenerateSecret: true,
+      LogoutURLs: workbenchCognitoProps.websiteUrls,
+      PreventUserExistenceErrors: 'ENABLED',
+      IdTokenValidity: 60,
+      AccessTokenValidity: 5,
+      RefreshTokenValidity: 1440,
+      TokenValidityUnits: {
+        IdToken: 'minutes',
+        AccessToken: 'minutes',
+        RefreshToken: 'minutes'
+      },
+      ExplicitAuthFlows: [
+        'ALLOW_USER_PASSWORD_AUTH',
+        'ALLOW_ADMIN_USER_PASSWORD_AUTH',
```

Comment:
should the web UI client have `ALLOW_ADMIN_USER_PASSWORD_AUTH`?

Reply:
I thought we discussed this on Slack. No, we shouldn't, but we need it until E2E tests are able to verify the settings. Should be removed once the E2E tests are set up.

Reply:
`ALLOW_ADMIN_USER_PASSWORD_AUTH` is not required for the web client due to `ALLOW_USER_PASSWORD_AUTH` allowing for the same tests.

Reply:
There are two tests in this file, one is `correctly uses default values` and this one. In first one (line 89) it correctly populates the array with only 
```
ExplicitAuthFlows: [
        'ALLOW_USER_PASSWORD_AUTH',
        'ALLOW_CUSTOM_AUTH',
        'ALLOW_USER_SRP_AUTH',
        'ALLOW_REFRESH_TOKEN_AUTH'
      ]
```
but in this second one it generates also `ALLOW_ADMIN_USER_PASSWORD_AUTH`. I assumed it's because additional properties are being passed into the function but I'd like to hear your opinion.

Reply:
I didnt see any parameters that would allow `ALLOW_ADMIN_USER_PASSWORD_AUTH`, which is why I asked.

Reply:
From the way it was written I didn't realize that lodash [merge](https://lodash.com/docs/4.17.15#merge) modifies first parameter so any change is actually propagated to default values which I think is incorrect behavior so I changed both this and `userPoolProps` to first make a deep copy of object to preserve default values. 
Searching through the codebase it looks like `merge` is also being used in `workbench-core/audit/src/plugins/baseAuditPlugin.ts` which will modify data in parameter `auditEntry` - this might be something that consumers are not expecting (@htztomic )